### PR TITLE
inject: add PS1 crystal data

### LINKED ISSUE #1506: Restore PS1 purple colouring on save crystals
Now that we have the reflections so truthfully added to TR1X, the only thing missing about the crystals is their characteristic purple colour.

https://github.com/user-attachments/assets/e090a307-d9dd-447c-9af2-4bb312b689d2

@walkawayy I believe you'd identified the shade pretty precisely when doing the PS1 style menu UI, no? Else we can help.

### PR DESCRIPTION
**Pull request by lahm86**
Resolves #1506.

#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

~~This allows injections to replace an unused palette slot with a specific color, and so allows an alternative to matching to the existing level palette. Data provided for each case of the savegame crystal, and the relevant config options updated/provided. The colour used is `#4040FF`.~~

~~I initially had in-game code to find the unused palette slots, but it was considerably expensive on loading times, so instead opted for the individual injection files with the information pre-calculated.~~

~~I also adjusted the teleport command to check crystal status to avoid tp'ing to ones that are consumed, mainly to allow checking how the crystal reacts to the lighting in different areas of the level.~~

This adds an option to replace the default savegame crystal model with one that matches PS1, using palette colour `#4040FF` on the mesh.

### CODE REVIEW

**Review by aredfan — APPROVED** [review]

File: src/game/console_cmd.c [hunk 1]
```diff
@@ -182,7 +182,8 @@ static COMMAND_RESULT Console_Cmd_Teleport(const char *const args)
              item_num++) {
             const ITEM_INFO *const item = &g_Items[item_num];
             const bool is_pickup =
-                Object_IsObjectType(item->object_id, g_PickupObjects);
+                Object_IsObjectType(item->object_id, g_PickupObjects)
+                || item->object_id == O_SAVEGAME_ITEM;
```
Comment (rr-) [inline src/game/console_cmd.c]: This should be tracked in a separate issue + changelog entry. I suggest amending the condition below to spell `(is_pickup || item->object_id == O_SAVEGAME_ITEM)`, or rename this variable to something that reflects its purpose better (`is_consumable` for example).

**Review by rr- — COMMENTED** [review]
Let's hold on with this approach for now – I believe that rather than forcing the injected texture colors into the level palette, we really should start supporting true color textures.

**Comment by lahm86** [comment]
I've squashed out the console change and opened #1518 for that. Shall I just ditch this PR, or leave it open for reference for the time being?

**Comment by rr-** [comment]
@lahm86 let's leave it open, I should be able to deliver the true color texturing today.

**Comment by rr-** [comment]
Should be easier to implement with #1520 :relieved:  Our options include adding a new function to append a palette color, performing a texture swap, or ideally just injecting the entire object.

**Comment by lahm86** [comment]
Updated to make use of the new palette logic - so much easier!

**Review by aredfan — APPROVED** [review]
I've checked one crystal per level, and all are showing the right colour as expected whether the purple tint option is enabled or disabled.

**Review by rr- — APPROVED** [review]

**Pull request merged**